src/posix/platform/netif.cpp

```diff
@@ -1339,7 +1355,14 @@ static void processNetifAddrEvent(otInstance *aInstance, struct nlmsghdr *aNetli
                     netAddr.mAddress      = addr;
                     netAddr.mPrefixLength = ifaddr->ifa_prefixlen;
 
-                    error = otIp6AddUnicastAddress(aInstance, &netAddr);
+                    if (otIp6HasUnicastAddress(aInstance, &netAddr.mAddress))
+                    {
+                        error = OT_ERROR_NONE;
+                    }
+                    else
+                    {
+                        error = otIp6AddUnicastAddress(aInstance, &netAddr);
+                    }
```

Comment:
Here is my suggestion:
```suggestion
                    error = otIp6HasUnicastAddress(aInstance, &netAddr.mAddress);
                    
                    if (error == OT_ERROR_INVALID_ARGS)
                    {
                        error = OT_ERROR_NONE;
                    }
```